src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -15,64 +16,45 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     /// <summary>
     /// A TupleTypeSymbol represents a tuple type, such as (int, byte) or (int a, long b).
     /// </summary>
-    internal sealed class TupleTypeSymbol : NamedTypeSymbol
+    internal sealed class TupleTypeSymbol : WrapperNamedTypeSymbol
     {
-        private readonly NamedTypeSymbol _underlyingType;
-        private readonly ImmutableArray<TupleFieldSymbol> _fields;
-        private readonly bool _hasFriendlyNames;
+        private readonly ImmutableArray<Location> _locations;
+        private readonly ImmutableArray<Location> _elementLocations;
+        private readonly ImmutableArray<string> _elementNames;
+        private readonly ImmutableArray<TypeSymbol> _elementTypes;
+        private ImmutableArray<Symbol> _lazyMembers;
+        private SmallDictionary<Symbol, Symbol> _lazyUnderlyingDefinitionToMemberMap;
 
         internal const int RestPosition = 8; // The Rest field is in 8th position
+        internal const string TupleTypeName = "ValueTuple";
 
-        /// <summary>
-        /// Construct a TupleTypeSymbol from it's underlying ValueTuple type and element names.
-        ///
-        /// Parameter elementsNames has to empty or fully populated.
-        /// </summary>
-        private TupleTypeSymbol(
-            NamedTypeSymbol underlyingType,
-            ImmutableArray<string> elementNames,
-            AssemblySymbol accessWithin)
+        public TupleTypeSymbol(Location locationOpt, NamedTypeSymbol underlyingType, ImmutableArray<Location> elementLocations, ImmutableArray<string> elementNames, ImmutableArray<TypeSymbol> elementTypes)
```

Comment:
Consider ordering parameters by importance. `elementTypes` should not be last and should not be after `elementLocations` or `elementNames`.